projects/plugins/jetpack/extensions/blocks/button/edit.js

```diff
@@ -45,20 +45,25 @@ export function ButtonEdit( props ) {
 
 	const blockClasses = classnames( 'wp-block-button', className );
 
-	const buttonClasses = classnames( 'wp-block-button__link', {
-		'has-background': backgroundColor.color || gradientValue,
-		[ backgroundColor.class ]: ! gradientValue && backgroundColor.class,
-		'has-text-color': textColor.color,
-		[ textColor.class ]: textColor.class,
-		[ gradientClass ]: gradientClass,
-		'no-border-radius': 0 === borderRadius,
-		'has-custom-width': !! width,
-	} );
+	const buttonClasses = classnames(
+		'wp-block-button__link',
+		{
+			'has-background': backgroundColor.color || gradientValue,
+			[ backgroundColor.class ]: ! gradientValue && backgroundColor.class,
+			'has-text-color': textColor.color,
+			[ textColor.class ]: textColor.class,
+			[ gradientClass ]: gradientClass,
+			'no-border-radius': 0 === borderRadius,
+			'has-custom-width': !! width,
+		},
+		fontSize && { [ `has-${ fontSize }-font-size` ]: true }
+	);
 
 	const buttonStyles = {
 		...( ! backgroundColor.color && gradientValue
 			? { background: gradientValue }
 			: { backgroundColor: backgroundColor.color } ),
+		fontSize: attributes.style?.typography?.fontSize,
```

Comment:
I don't understand this. In the line 27 above we're getting the font size from `attributes.fontSize` and here we're getting it from `attributes.style.typography.fontSize`. Do we have two different attributes?

Reply:
It depends on how you configure the font size. You can either choose from a small, medium, large and extra-large list or manually choose a specific font size. Each of these methods stores the font size in a different location.
When we use the list we must add a button class, when we use the input field we add a style.